README.md

```diff
@@ -0,0 +1,560 @@
+    indexed: false
+    type: uint256
+  - name: shares
+    indexed: false
+    type: uint256
+```
+
+#### Withdraw
+
+`sender` has exchanged `shares` for `assets`, and transferred those `assets` to `receiver`.
+
+MUST be emitted when shares are withdrawn from the Vault in `ERC4626.redeem` or `ERC4626.withdraw` methods.
+
+```yaml
+- name: Withdraw
+  type: event
+
+  inputs:
+  - name: sender
+    indexed: true
+    type: address
+  - name: receiver
+    indexed: true
+    type: address
+  - name: assets
+    indexed: false
+    type: uint256
+  - name: shares
+    indexed: false
+    type: uint256
+```
+
+## Rationale
+
+The Vault interface is designed to be optimized for integrators with a feature complete yet minimal interface.
+Details such as accounting and allocation of deposited tokens are intentionally not specified,
+as Vaults are expected to be treated as black boxes on-chain and inspected off-chain before use.
+
+ERC-20 is enforced because implementation details like token approval
+and balance calculation directly carry over to the shares accounting.
+This standardization makes the Vaults immediately compatible with all ERC-20 use cases in addition to ERC-4626.
+
+The mint method was included for symmetry and feature completeness.
+Most current use cases of share-based Vaults do not ascribe special meaning to the shares such that
+a user would optimize for a specific number of shares (mint) rather than specific amount of underlying (deposit).
+However, it is easy to imagine future Vault strategies which would have unique and independently useful share representations.
+
+A single `assetsPerShare` method can only be guaranteed to be exact with one of the four mutable methods,
+unless significant conditions are placed on the use cases that can comply with this standard.
+Use cases that require to know the value of a Vault position need to know the result of a `redeem` call, without executing it.
+On the other hand, integrators that intend to call `withdraw` on Vaults with the user approving only the exact amount of
+underlying need the result of a `withdraw` call. Similar use cases can be found for `deposit` and `mint`.
+
+As such, the `assetsPerShare` method has been kept for ease of integration on part of the simpler use cases,
+but `preview*` methods have been included for each one of the four mutable methods.
+In each case, the value of a preview method is only guaranteed to equal the return value of the relted mutable method
+if called immediately before in the same transaction.
+
+The `max*` methods are used to check for deposit/withdraw limits on vault capacity. These can be consumed off-chain for more user focused applications or on-chain for more on-chain aggregation/integration use cases.
+
+If implementors intend to support EOA account access directly, they should consider adding an additional function with the means to revert due to slippage loss since they are unable to revert if the output is undesirable.
```

Comment:
```suggestion
If implementors intend to support EOA account access directly, they should consider adding an additional function with the means to accomodate slippage loss or deposit/withdrawal limits,
since their transaction will revert if the exact amount is not acheived.
```